### PR TITLE
MULE-10404 - Fix Error API issues

### PR DESCRIPTION
**Pull request by pablolagreca**

### CODE REVIEW

File: core-tests/src/test/java/org/mule/runtime/core/execution/MessageProcessorNotificationExecutionInterceptorTestCase.java [hunk 1]
```diff
@@ -19,7 +20,17 @@
 import static org.mule.runtime.core.DefaultMuleEvent.getCurrentEvent;
 import static org.mule.runtime.core.DefaultMuleEvent.setCurrentEvent;
 
-import org.mule.runtime.core.exception.MessagingException;
+import java.util.ArrayList;
```
Comment (dfeist) [inline core-tests/src/test/java/org/mule/runtime/core/execution/MessageProcessorNotificationExecutionInterceptorTestCase.java]: You still have inverse import order to existing code and others..

File: modules/db/src/test/java/org/mule/runtime/module/db/internal/processor/DynamicBulkUpdateMessageProcessorTestCase.java [hunk 2]
```diff
@@ -75,6 +77,7 @@
   public void before() {
     when(event.getFlowCallStack()).thenReturn(new DefaultFlowCallStack());
     when(event.getExchangePattern()).thenReturn(REQUEST_RESPONSE);
+    when(event.getError()).thenReturn(Optional.empty());
```
Comment (afelisatti) [inline modules/db/src/test/java/org/mule/runtime/module/db/internal/processor/DynamicBulkUpdateMessageProcessorTestCase.java]: static import

File: core/src/main/java/org/mule/runtime/core/exception/OnErrorContinueHandler.java [hunk 3]
```diff
@@ -6,9 +6,12 @@
  */
 package org.mule.runtime.core.exception;
 
+import static java.util.Optional.empty;
```
Comment (dfeist) [inline core/src/main/java/org/mule/runtime/core/exception/OnErrorContinueHandler.java]: Are these imports need, no code changes.

File: tests/integration/src/test/java/org/mule/test/tck/ResponseAssertionMessageProcessorTestCase.java [hunk 4]
```diff
@@ -15,6 +16,8 @@
 import org.mule.tck.SensingNullMessageProcessor;
 import org.mule.tck.processor.TestNonBlockingProcessor;
 
+import java.util.Optional;
```
Comment (dfeist) [inline tests/integration/src/test/java/org/mule/test/tck/ResponseAssertionMessageProcessorTestCase.java]: Not required?

File: core-tests/src/test/java/org/mule/runtime/core/execution/TransactionalErrorHandlingExecutionTemplateTestCase.java [hunk 5]
```diff
@@ -26,7 +27,13 @@
 import static org.mule.runtime.core.execution.TransactionalErrorHandlingExecutionTemplate.createScopeExecutionTemplate;
 import static org.mule.runtime.core.transaction.TransactionTemplateTestUtils.getFailureTransactionCallbackStartsTransaction;
 
-import org.mule.runtime.core.exception.MessagingException;
+import org.junit.Before;
```
Comment (dfeist) [inline core-tests/src/test/java/org/mule/runtime/core/execution/TransactionalErrorHandlingExecutionTemplateTestCase.java]: Inverted imports

File: core/src/main/java/org/mule/runtime/core/DefaultMuleEvent.java [hunk 6]
```diff
@@ -6,9 +6,20 @@
  */
 package org.mule.runtime.core;
 
+import static java.util.Optional.ofNullable;
 import static org.mule.runtime.core.message.Correlation.NO_CORRELATION;
 import static org.mule.runtime.core.util.SystemUtils.getDefaultEncoding;
 
+import java.io.IOException;
```
Comment (dfeist) [inline core/src/main/java/org/mule/runtime/core/DefaultMuleEvent.java]: Inverted imports

File: modules/db/src/test/java/org/mule/runtime/module/db/internal/processor/DynamicBulkUpdateMessageProcessorTestCase.java (hunk 2 above)
Comment (pablolagreca) [inline modules/db/src/test/java/org/mule/runtime/module/db/internal/processor/DynamicBulkUpdateMessageProcessorTestCase.java]: Can't use it. Matchers.empty() already is using static imports

**Pull request merged**